api/src/ethereum/parent_mmr_root.rs

```diff
@@ -0,0 +1,37 @@
+//! Ethereum MMR API
+use mmr::{Database, build_client};
+use actix_web::{
+    web::{Data, Path, Json},
+    Responder
+};
+use crate::{Result, Error, AppData};
+use serde::Serialize;
+
+/// MMR root result
+#[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum MMRRootResult {
+    MmrRoot { mmr_root: String },
+    Error { error: String }
+}
+
+/// Get target mmr
+#[allow(clippy::eval_order_dependence)]
+pub async fn handle(block: Path<String>, app_data: Data<AppData>) -> impl Responder {
+    match mmr_root(block, &app_data.mmr_db) {
+        Ok(root) => Json(MMRRootResult::MmrRoot { mmr_root: format!("0x{}", root) }),
+        Err(err) => Json(MMRRootResult::Error { error: err.to_string() })
+    }
+}
+
+fn mmr_root(block: Path<String>, mmr_db: &Database) -> Result<String> {
+    let block: u64 = block.to_string().parse()?;
+    let leaf_index = block - 1;
+    let client = build_client(mmr_db)?;
+    let result = client.get_mmr_root(leaf_index)?;
```

Comment:
get `mmr_root` or `parent_mmr_root`?

Reply:
Reading from db?

Reply:
It depends on the implementation of mmr client's get_mmr_root function.
If MySQL, it is from calc now. It was from MySQL directly. But for the import performance, mmr root is not generated when import.

Reply:
I plan to add an async task to run mmr_root generation for MySQL. If already exist in the db, just get from db and return it. If not exist, calc and save to db.

Reply:
> I plan to add an async task to run mmr_root generation for MySQL. If already exist in the db, just get from db and return it. If not exist, calc and save to db.

`calc now` looks good for me if have no performance bottleneck. Prefer to keep simple instead of adding extra process logic.